packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx

```diff
@@ -111,63 +99,78 @@ const DetailsFormStep = ({
     setPayoutStructure((prevState) => [...prevState.slice(0, -1)]);
   };
 
-  const handleToggleTopic = (topicId: number) =>
-    setToggledTopicList((prevState) => {
-      const isChecked = prevState.find((topic) => topic.id === topicId).checked;
-
-      return prevState.map((topic) => {
-        if (topic.id === topicId) {
-          return {
-            ...topic,
-            checked: !isChecked,
-          };
-        }
-        return topic;
-      });
-    });
-
-  const handleToggleAllTopics = () => {
-    const mappedTopics = sortedTopics?.map((topic) => ({
-      ...topic,
-      checked: !allTopicsToggled,
-    }));
-    setToggledTopicList(mappedTopics);
-    setAllTopicsToggled((prevState) => !prevState);
-  };
-
   const goBack = () => {
     navigate('/manage/contests');
   };
 
   const handleCancel = () => {
-    // TODO show warning that prevents quitting
-    goBack();
+    openConfirmation({
+      title: 'You have unsaved changes',
+      description:
+        'Are you sure you want to exit? Your contest will not be launched and your changes will not be saved.',
+      buttons: [
+        {
+          label: 'Cancel',
+          buttonType: 'secondary',
+          buttonHeight: 'sm',
+        },
+        {
+          label: 'Exit',
+          buttonType: 'destructive',
+          buttonHeight: 'sm',
+          onClick: goBack,
+        },
+      ],
+    });
   };
 
-  const handleSubmit = (values: ContestFormSubmitValues) => {
-    console.log('values', values);
+  const handleSubmit = (values: ContestFormValidationSubmitValues) => {
```

Comment:
i found that zod provides typescript types for subimt values, we can do
```
values: z.infer<typeof detailsFormValidationSchema>,
```

Reply:
this is exactly what I do but just exported as a type in [this file](https://github.com/hicommonwealth/commonwealth/blob/f979a0dd795f842eb6cf16f55f29ed82ab509387/packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/types.ts#L19)

<img width="460" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/d36096d3-880b-4b41-a69e-e049bfc7533b">

Reply:
oh nice. Didn't notice that. Thanks for explaination